clients/python/test/e2e/ci/setup_e2e_pytest.py

```diff
@@ -0,0 +1,121 @@
+import toml
+from pathlib import Path
+import pandas as pd
+from urllib.parse import urlparse, ParseResult
+from typing import List, Dict, Any, Union
+import typer
+from packaging import version
+import json
+from _utils import E2eScriptFailure, E2eExitCodes
+import warnings
+
+# keys to be found in input dicts
+surl: str = "OSPARC_API_HOST"
+skey: str = "OSPARC_API_KEY"
+ssecret: str = "OSPARC_API_SECRET"
+skeys: List[str] = [surl, skey, ssecret]
+
+crepo: str = "OSPARC_CLIENT_REPO"
+cbranch: str = "OSPARC_CLIENT_BRANCH"
+cversion: str = "OSPARC_CLIENT_VERSION"
+ckeys: List[str] = [crepo, cbranch, cversion]
+
+
+def main(client_config: str, server_config: str) -> bool:
+    """
+    Generates a toml configuration file pytest e2e tests
+
+    exceptions:
+    -----------
+        A typer.Exit(code=100) is raised if a failure is encountered
+
+    returns:
+    --------
+        A bool indicating whether or not the (client, server) pair are compatible
+    """
+    # read in data
+    ccfg = json.loads(client_config)
+    scfg = json.loads(server_config)
+    if not isinstance(ccfg, dict):
```

Comment:
TIP: here you could create a little pydantic class and use it to
a) parse from json-file the configuration
b) validate it

From line37 to 58 wjould be done in two lines and more reliably :-)

Reply:
Yes, this is a great idea. I have implemented this.